daffodil-tdml-processor/src/test/scala/org/apache/daffodil/processor/tdml/TestDaffodilC.scala

```diff
@@ -137,7 +137,7 @@ class TestDaffodilC {
     val input = new ByteArrayInputStream(b)
     val pr = tdp.parse(input, b.length * 8)
     assert(
-      !pr.isProcessingError && pf.getDiagnostics.isEmpty,
+      !pr.isProcessingError,
```

Comment:
Let's discuss whether to remove `&& pf.getDiagnostics.isEmpty` or not.  When we compile the generated C code into an executable and run that executable, the program will invariably print validation diagnostics if it detects validation problems like an element's value not matching its fixed attribute.  However, the program returns an error exit status only if you pass `-V on` or `-V limited`, otherwise it returns a successful exit status.  I wanted to fail these test cases and print the validation diagnostics if any are printed.

Note that right now, the TDML Runner doesn't pass any validation option to the daffodilC executable, but the TDML Runner does return any printed messages as validation diagnostics in the TDML result, so that the TDML Runner will do the right thing with any validation error clauses in the test cases.

Reply:
I originally removed this because of the new warnings about a missing leading slash that causes this to fail (which are fixed in the subsequent commit in this PR). So there were new schema compile-time diagnostics that this check didn't like.

> the program will invariably print validation diagnostics if it detects validation problems like an element's value not matching its fixed attribute

Are these parse-time diagnostics that you expect to be empty? Not schema compile/code generation-time diagnostics? Maybe this should have be `pr.getDiagnostics.isEmpty` instead of `pf.getDiagnostics.isEmpty`? Would that be sufficient for the check you're looking for, since the new warnings should only be on the ProcessorFactory but not on the ParseResult (I think?).

Reply:
Yes, you're right, `pf.getDiagnostics.isEmpty` should be `pr.getDiagnostics.isEmpty` (parse-time diagnostics, not schema compile/code generation-time diagnostics).